src/main/drivers/rx_nrf24l01.c

```diff
@@ -0,0 +1,346 @@
+{
+    if(mode == TX_EN) {
+        NRF24_CE_LO();
+        NRF24L01_WriteReg(NRF24L01_07_STATUS, (1 << NRF24L01_07_STATUS_RX_DR)    //reset the flag(s)
+                                            | (1 << NRF24L01_07_STATUS_TX_DS)
+                                            | (1 << NRF24L01_07_STATUS_MAX_RT));
+        NRF24L01_WriteReg(NRF24L01_00_CONFIG, (1 << NRF24L01_00_CONFIG_EN_CRC)   // switch to TX mode
+                                            | (1 << NRF24L01_00_CONFIG_CRCO)
+                                            | (1 << NRF24L01_00_CONFIG_PWR_UP));
+        delayMicroseconds(130);
+        NRF24_CE_HI();
+    } else if (mode == RX_EN) {
+        NRF24_CE_LO();
+        NRF24L01_WriteReg(NRF24L01_07_STATUS, 0x70);        // reset the flag(s)
+        NRF24L01_WriteReg(NRF24L01_00_CONFIG, 0x0F);        // switch to RX mode
+        NRF24L01_WriteReg(NRF24L01_07_STATUS, (1 << NRF24L01_07_STATUS_RX_DR)    //reset the flag(s)
+                                            | (1 << NRF24L01_07_STATUS_TX_DS)
+                                            | (1 << NRF24L01_07_STATUS_MAX_RT));
+        NRF24L01_WriteReg(NRF24L01_00_CONFIG, (1 << NRF24L01_00_CONFIG_EN_CRC)   // switch to RX mode
+                                            | (1 << NRF24L01_00_CONFIG_CRCO)
+                                            | (1 << NRF24L01_00_CONFIG_PWR_UP)
+                                            | (1 << NRF24L01_00_CONFIG_PRIM_RX));
+        delayMicroseconds(130);
+        NRF24_CE_HI();
+    } else {
+        NRF24L01_WriteReg(NRF24L01_00_CONFIG, (1 << NRF24L01_00_CONFIG_EN_CRC)); //PowerDown
+        NRF24_CE_LO();
+    }
+}
+
+bool NRF24L01_Reset()
+{
+    NRF24L01_SetTxRxMode(TXRX_OFF);
+    return NRF24L01_ReadReg(0x07) == 0x0E;
+}
+
+/*
+ * Enter standby mode
+ */
+void NRF24L01_SetStandbyMode(void)
+{
+    // set CE low and clear the PRIM_RX bit to enter standby mode
+    NRF24_CE_LO();
+    NRF24L01_WriteReg(NRF24L01_00_CONFIG, standbyConfig);
+}
+
+/*
+ * Enter receive mode
+ */
+void NRF24L01_SetRxMode(void)
+{
+    NRF24_CE_LO(); // drop into standby mode
+    // set the PRIM_RX bit
+    NRF24L01_WriteReg(NRF24L01_00_CONFIG, standbyConfig | BV(NRF24L01_00_CONFIG_PRIM_RX));
+    NRF24L01_ClearAllInterrupts();
+    // finally set CE high to start enter RX mode
+    NRF24_CE_HI();
+    // nRF24L01+ needs 130 microseconds settling time from Standby mode to RX mode
+    delayMicroseconds(130);
+}
+
+/*
+ * Enter transmit mode. Anything in the transmit FIFO will be transmitted.
+ */
+void NRF24L01_SetTxMode(void)
+{
+    // Ensure in standby mode, since can only enter TX mode from standby mode
+    NRF24L01_SetStandbyMode();
+    NRF24L01_ClearAllInterrupts();
+    // pulse CE for 10 microseconds to enter TX mode
+    NRF24_CE_HI();
+    delayMicroseconds(10);
+    NRF24_CE_LO();
+    // nRF24L01+ needs 130 microseconds settling time from Standby mode to TX mode
+    delayMicroseconds(130);
```

Comment:
Again - after 130us transmission will start. But it will take some time to transmit the data, co caller need to wait for end of transmission anyway.